openfisca_uk/variables/finance/benefit/family/universal_credit.py

```diff
@@ -1,134 +1,359 @@
     entity = BenUnit
-    label = u"Personal allowance for Universal Credit"
+    label = u"UC child element"
     definition_period = YEAR
 
     def formula(benunit, period, parameters):
         UC = parameters(period).benefit.universal_credit
-        is_single = benunit("is_single", period)
-        is_couple = benunit("is_couple", period)
-        is_over_25 = benunit("youngest_adult_age", period.this_year) >= 25
-        basic_amount = (
-            is_single * not_(is_over_25) * UC.allowances.basic_single_young
-            + is_single * is_over_25 * UC.allowances.basic_single_older
-            + is_couple * not_(is_over_25) * UC.allowances.basic_couple_young
-            + is_couple * is_over_25 * UC.allowances.basic_couple_older
-        ) * MONTHS_IN_YEAR
-        return basic_amount
+        return benunit.sum(
+            benunit.members("UC_individual_child_element", period)
+        )
 
 
-class UC_premiums(Variable):
+class UC_carer_element(Variable):
     value_type = float
     entity = BenUnit
-    label = u"Premiums for Universal Credit"
+    label = u"UC carer element"
     definition_period = YEAR
 
     def formula(benunit, period, parameters):
         UC = parameters(period).benefit.universal_credit
         has_carer = (
             aggr(benunit, period.this_year, ["is_carer_for_benefits"]) > 0
         )
-        num_eligible_children = min_(2, benunit.nb_persons(BenUnit.CHILD))
-        childcare_limit = (
-            num_eligible_children == 1
-        ) * UC.elements.max_childcare_1 * MONTHS_IN_YEAR + (
-            num_eligible_children > 1
-        ) * UC.elements.max_childcare_2 * MONTHS_IN_YEAR
-        premiums = (
-            has_carer * UC.elements.carer_element
-            + num_eligible_children * UC.elements.child_element
-            + benunit("num_disabled_children", period.this_year)
-            * UC.elements.disabled_child_element
-            + benunit("num_disabled_adults", period.this_year)
-            * UC.elements.disabled_element
-        ) * MONTHS_IN_YEAR
-        eligible_childcare_costs = benunit.sum(
-            benunit.members("childcare_cost", period)
+        return UC.elements.carer.amount * has_carer * MONTHS_IN_YEAR
+
+
+class UC_housing_costs_element(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"UC housing costs element"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        tenure_type = benunit.value_from_first_person(
+            benunit.members.household("tenure_type", period).decode_to_str()
         )
-        childcare_element = min_(
-            childcare_limit,
-            eligible_childcare_costs * UC.elements.childcare_cost_rate,
+        max_housing_costs = select(
+            [
+                (tenure_type == TenureType.RENT_FROM_COUNCIL.name)
+                | (tenure_type == TenureType.RENT_FROM_HA.name),
+                tenure_type == TenureType.RENT_PRIVATELY.name,
+                True,
```

Comment:
would `.fillna(0)` at the end also work?

Reply:
I think the select function needs a condition every time.